components/inx/server_commitments.go

```diff
@@ -46,3 +50,61 @@ func (s *Server) ReadCommitment(_ context.Context, req *inx.CommitmentRequest) (
 
 	return inxCommitment(commitment), nil
 }
+
+func (s *Server) ListenToLatestCommitments(_ *inx.NoParams, srv inx.INX_ListenToLatestCommitmentsServer) error {
+	ctx, cancel := context.WithCancel(Component.Daemon().ContextStopped())
+
+	wp := workerpool.New("ListenToCommitments", workerCount).Start()
+
+	unhook := deps.Protocol.Events.Engine.Notarization.SlotCommitted.Hook(func(commitmentDetails *notarization.SlotCommittedDetails) {
+		inxCommitment := inx.NewCommitmentWithBytes(commitmentDetails.Commitment.ID(), commitmentDetails.Commitment.Data())
+
+		if err := srv.Send(inxCommitment); err != nil {
+			Component.LogErrorf("send error: %v", err)
+			cancel()
+		}
+	}, event.WithWorkerPool(wp)).Unhook
+
+	<-ctx.Done()
+	unhook()
+
+	// We need to wait until all tasks are done, otherwise we might call
+	// "SendMsg" and "CloseSend" in parallel on the grpc stream, which is
+	// not safe according to the grpc docs.
+	wp.Shutdown()
+	wp.ShutdownComplete.Wait()
+
+	return ctx.Err()
+}
+
+func (s *Server) ListenToFinalizedCommitments(_ *inx.NoParams, srv inx.INX_ListenToFinalizedCommitmentsServer) error {
```

Comment:
I'm wondering if we need the possibility to send previous finalized commitments before we actually start sending the new ones. At least we needed that in hornet for the milestones.
There we had this whole `rangeSend` magic (see `ListenToConfirmedMilestones`). Maybe you can have a look and we can discuss if we should add this in a follow up PR?